src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,185 @@
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        int newSize = size + list.size();
+        if (newSize < array.length) {
+            copyArray(list);
+        } else {
+            checkIfArrayShouldGrow(list);
+            copyArray(list);
+        }
 
+        size += list.size();
     }
 
     @Override
-    public T get(int index) {
-        return null;
+    public T get(int index) throws ArrayListIndexOutOfBoundsException {
+        checkIfIndexNegative(index);
+        checkIfPositionExistent(index);
+        T result = (T) array[index];
+        return result;
     }
 
     @Override
-    public void set(T value, int index) {
-
+    public void set(T value, int index) throws ArrayListIndexOutOfBoundsException {
+        checkIfIndexNegative(index);
+        checkIfPositionExistent(index);
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIfIndexNegative(index);
+        checkIfPositionExistent(index);
+        Object[] result = Arrays.copyOf(array, array.length - 1);
+        T removedItem = null;
+        removedItem = (T) array[index];
+        System.arraycopy(array, index + 1, result, index, array.length - index - 1);
+        array = result;
+        size--;
+        return removedItem;
     }
 
     @Override
-    public T remove(T element) {
+    public T remove(T element) throws NoSuchElementException {
+        boolean flag = false;
+
+        for (int i = 0; i < size; i++) {
+            if (array[i] == null) {
+                if (element == null) {
+                    remove(getIndex(element));
+                    flag = true;
+                    return element;
+                }
+                continue;
+            }
+            if (array[i] == element || array[i].equals(element)) {
+                remove(getIndex(element));
+                flag = true;
+                return element;
+            }
+
+        }
+        if (flag == false) {
+            throw new NoSuchElementException("The element is not present in the list");
+        }
         return null;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    public void grow() {
+        Object[] result = new Object[(int) (array.length * 1.5) + 1];
+        System.arraycopy(array, 0, result, 0, array.length);
+        array = result;
+    }
+
+    public void addElement(T value) {
+        array[size] = value;
+    }
+
+    public void addElement(T value, int position) {
+        Object[] result = new Object[array.length];
+        System.arraycopy(array, 0, result, 0, position);
+        result[position] = value;
+        System.arraycopy(array, position, result, position + 1, size - position);
+        array = result;
+    }
+
+    public void copyArray(List<T> list) {
```

Comment:
Don't begin method implementation with an empty line. Remove the empty line at the beginning of the method.